### PR TITLE
fqdn: Make maximum number of IPs per restored rule configurable

### PR DESCRIPTION
**Pull request by jrajahalme**
Only count the number of IPs for each FQDN selector/rule when storing
rules for restoration, rather than ignoring later rules on a port
after previous rules have hit the maximum number of IPs.

Make the maximum number of IPs per restored rule configurable with the
new option '--tofqdns-max-ips-per-restored-rule' (default 1000).

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
FQDN rule restoration IP limit has been made configurable (`--tofqdns-max-ips-per-restored-rule`, default 1000). 
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-backport-1.7

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
gke build fails due to an ald assumption in  select cluster script which was not updated on 1.7 branch. I will push a commit which fixes that. for posterity, upstream test, missed-k8s and docs builds passed

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
This PR is currently hitting docker hub image pull rate limiting, which are mitigated in https://github.com/cilium/cilium/pull/13959 and https://github.com/cilium/cilium/pull/13969 (the latter backport is pending, the former not yet added to backport PR).

File: Documentation/cmdref/cilium-agent.md [hunk 1]
```diff
@@ -168,6 +168,7 @@ cilium-agent [flags]
       --tofqdns-enable-poller-events                  Emit DNS responses seen by the DNS poller as Monitor events, if the poller is enabled. (default true)
       --tofqdns-endpoint-max-ip-per-hostname int      Maximum number of IPs to maintain per FQDN name for each endpoint (default 50)
       --tofqdns-max-deferred-connection-deletes int   Maximum number of IPs to retain for expired DNS lookups with still-active connections (default 10000)
+      --tofqdns-max-ips-per-restored-rule int                Maximum number of IPs to maintain for each restored rule (default 1000)
```
Comment (christarazi) [inline Documentation/cmdref/cilium-agent.md:171]: Please add the upstream commit to the commit msg, now that it has been merged.

**Review by christarazi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**